fbrp/src/fbrp/runtime/conda.py

```diff
@@ -148,20 +150,19 @@ async def death_handler(self):
 
     async def handle_down(self):
         try:
-            if self.proc.returncode is not None:
-                return
-            proc_pid = self.get_pid()
+            if self.proc.returncode is None:
+                proc_pid = self.get_pid()
 
-            life_cycle.set_state(self.name, life_cycle.State.STOPPING)
-            os.kill(proc_pid, signal.SIGTERM)
+                life_cycle.set_state(self.name, life_cycle.State.STOPPING)
+                os.kill(proc_pid, signal.SIGTERM)
```

Comment:
actually you may just want `self.proc.send_sginal(signal.SIGTERM)`

Reply:
Done